kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ContainerFileReference.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.nio.file.Path;
+import java.util.Objects;
+
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * A reference to a file in a container.
+ * @param volume The volume on which the file will exist
+ * @param mount The volume mount beneath which the file will exist
+ * @param containerPath The absolute path of the file
+ */
+public record ContainerFileReference(
+                                     @Nullable Volume volume,
+                                     @Nullable VolumeMount mount,
+                                     @NonNull Path containerPath) {
```

Comment:
Strictly speaking using Path might be a bit flaky, because I guess we maybe shouldn't assume we're running in a linux container, or that the OS for the operator is the same as the OS for the proxy, and therefore we don't know that the path syntax (and the directory separator in particular) in the operator JVM is going to be correct/the same as the proxy. But this definitely is a quibble I'm prepared to overlook for now.

Reply:
Path is good enough!  If we end up with demand for running on Windows, or something like OS/2 😊 etc, we can think about it then.